### PR TITLE
fix: yaml is correctly highlighted

### PR DESCRIPTION
**Pull request by deer**
closes https://github.com/denoland/deno-gfm/issues/60

Sadly prism doesn't offer an easy way to import languages, so I've just hacked it in by doing:
```ts
import "https://esm.sh/prismjs@1.29.0/components/prism-yaml";
```

The testing strategy I was pursuing needed a major refactor, which I've done here. It's now significantly easier to add server cases. Additionally, you can now easily visually inspect the test cases by running `deno task server`. This currently produces the following output:
```md
# Deno GFM Server Tests

[basicMarkdownTable](http://localhost:8000/basicMarkdownTable)
[yaml](http://localhost:8000/yaml)
```

I think the highlighting on the yaml case can be improved. Please let me know what I should be doing in `main.scss` to improve this.

### CODE REVIEW

File: mod.ts [hunk 1]
```diff
@@ -11,6 +11,7 @@ import {
 } from "./deps.ts";
 import { CSS, KATEX_CLASSES, KATEX_CSS } from "./style.js";
 export { CSS, KATEX_CSS, Marked };
+import "https://esm.sh/prismjs@1.29.0/components/prism-yaml";
```
Comment (kyeotic) [inline mod.ts:14]: Is pulling in yaml for everyone necessary? It seems like the actual fix is the additional types below. Won't that still work if the user app pulls in YAML (like it has to pull in most languages)?

File: style/main.scss [hunk 2]
```diff
@@ -61,6 +61,14 @@
     color: var(--color-prettylights-syntax-constant);
   }
 
+  &.atrule {
+    color: var(--color-prettylights-syntax-keyword);
+  }
+
+  &.punctuation {
```
Comment (kyeotic) [inline style/main.scss:68]: You added `key` to the options above, but didn't add a matching style. Was that intentional?

File: mod.ts (hunk 1 above)
Comment (deer) [reply mod.ts:14]: Sadly I think it is. The necessary thing is making sure this returns something:
https://github.com/denoland/deno-gfm/blob/b9f7352e618138d62201a245a9f6a44e23a2ba6d/mod.ts#L53

The only way I could do that is by importing the package. For whatever reason I couldn't find any way to invoke the loadlanguages function that prism has.

File: style/main.scss (hunk 2 above)
Comment (deer) [reply style/main.scss:68]: I think the css produced purely by prism is also lacking css for this, although I could be wrong. Is it the styling looking incorrect?

**Comment by deer** [comment]
Waiting on https://github.com/denoland/deno-gfm/pull/99, so that this PR is focused purely on YAML, instead of also redoing the testing infrastructure.

**Comment by deer** [comment]
@hashrock and @crowlKats, what do you think about this one? On the positive side, it highlights yaml better than we currently do. On the negative side, it might not be right in all cases and if the user doesn't want yaml highlighting, there's no way to opt out. I'd say the positives outweigh the negatives though.

**Comment by Mrashes** [comment]
@deer I agree positives outweigh the negatives.  It would be a positive addition to the package if added, also contributes to the goal of it all working out the box.

**Review by crowlKats — APPROVED** [review]
LGTM

**Pull request merged**